src/map/npc.c

```diff
@@ -1980,6 +1986,8 @@ static void npc_expanded_barter_tosql(struct npc_data *nd, int index)
  */
 static void npc_expanded_barter_delfromsql_sub(const char *npcname, int itemId, int zeny, int currencyCount, struct npc_barter_currency* currency)
 {
+	nullpo_retv(npcname);
+	nullpo_retv(currency);
```

Comment:
I believe this check should be only done in the else branch (the function is called with `itemId = INT_MAX, zeny = 0, currencyCount = 0, currency = NULL` in order to clear all)